File: modules/brl/native/databuffer.cs

public class BBDataBuffer{

    public byte[] _data;
    public int _length;
    
    public BBDataBuffer(){
    }

    public BBDataBuffer( int length ){
    	_data=new byte[length];
    	_length=length;
    }
    
    public BBDataBuffer( byte[] data ){
    	_data=data;
    	_length=data.Length;
    }
    
    public bool _New( int length ){
    	if( _data!=null ) return false;
    	_data=new byte[length];
    	_length=length;
    	return true;
    }
    
    public bool _Load( String path ){
    	if( _data!=null ) return false;
    	
    	_data=MonkeyData.loadBytes( path );
    	if( _data==null ) return false;
    	
    	_length=_data.Length;
    	return true;
    }
    
    public void Discard(){
    	if( _data!=null ){
    		_data=null;
    		_length=0;
    	}
    }
    
  	public int Length(){
  		return _length;
  	}

	public void PokeByte( int addr,int value ){
		_data[addr]=(byte)value;
	}

	public void PokeShort( int addr,int value ){
		Array.Copy( System.BitConverter.GetBytes( (short)value ),0,_data,addr,2 );
	}

	public void PokeInt( int addr,int value ){
		Array.Copy( System.BitConverter.GetBytes( value ),0,_data,addr,4 );
	}

	public void PokeFloat( int addr,float value ){
		Array.Copy( System.BitConverter.GetBytes( value ),0,_data,addr,4 );
	}

	public int PeekByte( int addr ){
		return (int)(sbyte)_data[addr];
	}

	public int PeekShort( int addr ){
		return (int)System.BitConverter.ToInt16( _data,addr );
	}

	public int PeekInt( int addr ){
		return System.BitConverter.ToInt32( _data,addr );
	}

	public float PeekFloat( int addr ){
		return (float)System.BitConverter.ToSingle( _data,addr );
	}
}

File: modules/brl/native/databuffer.java

class BBDataBuffer{

	boolean _New( int length ){
		if( _data!=null ) return false;
		_data=ByteBuffer.allocate( length );
		_data.order( ByteOrder.nativeOrder() );
		_length=length;
		return true;
	}
	
	boolean _Load( String path ){
		if( _data!=null ) return false;
	
		byte[] bytes=MonkeyData.loadBytes( path );
		if( bytes==null ) return false;
		
		int length=bytes.length;
		
		if( !_New( length ) ) return false;
		
		System.arraycopy( bytes,0,_data.array(),0,length );
		
		return true;
	}
	
	int Length(){
		return _length;
	}
	
	void Discard(){
		if( _data==null ) return;
		_data=null;
		_length=0;
	}
		
	void PokeByte( int addr,int value ){
		_data.put( addr,(byte)value );
	}
	
	void PokeShort( int addr,int value ){
		_data.putShort( addr,(short)value );
	}
	
	void PokeInt( int addr,int value ){
		_data.putInt( addr,value );
	}
	
	void PokeFloat( int addr,float value ){
		_data.putFloat( addr,value );
	}
	
	int PeekByte( int addr ){
		return _data.get( addr );
	}
	
	int PeekShort( int addr ){
		return _data.getShort( addr );
	}
	
	int PeekInt( int addr ){
		return _data.getInt( addr );
	}
	
	float PeekFloat( int addr ){
		return _data.getFloat( addr );
	}

	ByteBuffer GetByteBuffer(){
		return _data;
	}
	
	ByteBuffer _data;
	int _length;
}
